cocotb/share/lib/fli/FliObjHdl.cpp

```diff
@@ -473,13 +476,15 @@ int FliStringObjHdl::initialise(std::string &name, std::string &fq_name)
     m_num_elems   = mti_TickLength(m_val_type);
     m_indexable   = true;
 
-    m_mti_buff    = (char*)malloc(sizeof(char) * m_num_elems);
+    size_t amt    = static_cast<size_t>(m_num_elems);
+    m_mti_buff    = (char*)malloc(amt);
     if (!m_mti_buff) {
         LOG_CRITICAL("Unable to alloc mem for value object mti read buffer: ABORTING");
         return -1;
     }
 
-    m_val_buff = (char*)malloc(m_num_elems+1);
+    amt           = static_cast<size_t>(m_num_elems) + 1;
+    m_val_buff    = (char*)malloc(amt);
```

Comment:
Frankly we should just be using `m_val_buff = new char[amt];` here and everywhere else. I suppose that's out of scope though.

Reply:
Wait, they're C++ files? Couldn't tell...

If it makes warnings go away, I'll do it in this PR.